modules/accumulo/src/test/java/io/fluo/accumulo/iterators/GarbageCollectionIteratorTest.java

```diff
@@ -0,0 +1,456 @@
+    //TODO need to test w/ more types
+    input.add("0 f q TX_DONE 21", "");
+    input.add("0 f q WRITE 21", "20 TRUNCATION PRIMARY");
+    input.add("0 f q ACK 20", "");
+    input.add("0 f q ACK 16", "");
+    input.add("0 f q ACK 14", "");
+
+    TestData expected = new TestData();
+    expected.add("0 f q TX_DONE 21", "");
+    expected.add("0 f q WRITE 21", "20 TRUNCATION PRIMARY");
+
+    TestData output = new TestData(newGCI(input, 5));
+
+    Assert.assertEquals(2, expected.data.size());
+    Assert.assertEquals(expected, output);
+
+    input = new TestData();
+    input.add("0 f q TX_DONE 21", "");
+    input.add("0 f q ACK 20", "");
+    input.add("0 f q ACK 16", "");
+    input.add("0 f q ACK 14", "");
+
+    expected = new TestData();
+    expected.add("0 f q TX_DONE 21", "");
+    expected.add("0 f q ACK 20", "");
+
+    output = new TestData(newGCI(input, 5));
+
+    Assert.assertEquals(2, expected.data.size());
+    Assert.assertEquals(expected, output);
+
+    input = new TestData();
+    input.add("0 f q WRITE 19", "16 TRUNCATION PRIMARY");
+    input.add("0 f q ACK 20", "");
+    input.add("0 f q ACK 16", "");
+    input.add("0 f q ACK 14", "");
+
+    expected = new TestData();
+    expected.add("0 f q WRITE 19", "16 TRUNCATION PRIMARY");
+    expected.add("0 f q ACK 20", "");
+
+    output = new TestData(newGCI(input, 5));
+
+    Assert.assertEquals(2, expected.data.size());
+    Assert.assertEquals(expected, output);
+  }
+
+  @Test
+  public void testDataRemoval(){
+    TestData input = new TestData();
+
+    input.add("0 f q WRITE 52","48");
+    input.add("0 f q WRITE 45","41");
+    input.add("0 f q WRITE 38","33");
+    input.add("0 f q WRITE 20","17 TRUNCATION");
+    input.add("0 f q DATA 48","18");
+    input.add("0 f q DATA 41","15");
+    input.add("0 f q DATA 33","13");
+    input.add("0 f q DATA 17","11");
+
+    TestData output = new TestData(newGCI(input, 3));
+    Assert.assertEquals(8, input.data.size());
+    Assert.assertEquals(input, output);
+
+    //this time falls between the start and commit times of one of the transactions that wrote data
+    output = new TestData(newGCI(input, 35));
+    Assert.assertEquals(input, output);
+
+    TestData expected = new TestData();
+    expected.add("0 f q WRITE 52","48");
+    expected.add("0 f q WRITE 45","41");
+    expected.add("0 f q WRITE 38","33 TRUNCATION");
+    expected.add("0 f q DATA 48","18");
+    expected.add("0 f q DATA 41","15");
+    expected.add("0 f q DATA 33","13");
+
+    output = new TestData(newGCI(input, 39));
+    Assert.assertEquals(6, expected.data.size());
+    Assert.assertEquals(expected, output);
+
+    expected = new TestData();
+    expected.add("0 f q WRITE 52","48");
+    expected.add("0 f q WRITE 45","41 TRUNCATION");
+    expected.add("0 f q DATA 48","18");
+    expected.add("0 f q DATA 41","15");
+
+    output = new TestData(newGCI(input, 46));
+    Assert.assertEquals(4, expected.data.size());
```

Comment:
> Is this assertion necessary given the next statement?
When I first wrote the `TestData` class it had a bug in it and was not adding anything.  I added this code to test the test code.